testing/test_unit.py

```diff
@@ -70,6 +70,26 @@ def pytest_html_results_summary(prefix, summary, postfix, session):
     result.assert_outcomes(passed=1)
 
 
+def test_chdir(tmp_path, pytester):
+    pytester.makepyfile(
+        """
+        import pytest
+
+        @pytest.fixture
+        def changing_dir(tmp_path, monkeypatch):
+            monkeypatch.chdir(tmp_path)
+
+        def test_function(changing_dir):
+            pass
+    """
+    )
+    page = pytester.runpytest("--html", "reports/report.html")
+    assert page.ret == 0
+    assert (
+        f"Generated html report: file://{tmp_path / 'reports' / 'report.html'}"
+    ).replace("test_chdir0", "test_chdir1") in page.outlines[-2]
```

Comment:
What's going on here? Why is this replace necessary?

Reply:
It is because I want to refer to the `tmp_path` variable of the internal test named `test_function`, not the actual test named `test_chdir`.

Reply:
Though I don't know why it's not called `test_function0`?

Reply:
> Though I don't know why it's not called `test_function0`?

Exactly what I was expecting.

@RonnyPfannschmidt Any insight here?

Reply:
I got it, it was because of the `tmp_path` fixture that was creating the `test_chdir0` folder and `pytester` was creating the `test_chdir1` folder. Now I directly use the path from `pytester` so it's simpler.

Reply:
Awesome!